diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt

```diff
@@ -461,7 +462,7 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
 
         if (psi.children.isNotEmpty() && !psi.isFirstChildElementType(DOT_QUALIFIED_EXPRESSION) &&
                 !psi.isFirstChildElementType(SAFE_ACCESS_EXPRESSION)) {
-            val firstChild = psi.firstChild
+            val firstChild = if (psi.isFirstChildElementType(ARRAY_ACCESS_EXPRESSION)) psi.firstChild.firstChild else psi.firstChild
```

Comment:
Is that just workaround?

I looked at this bug before and think, that the main problem is in these lines:

https://github.com/cqfn/diKTat/blob/37b2478924f69d9f4be205bc46a2a12bbed89fb8/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt#L469-L471

For now I didn't catch the logic of this `if` statement - in which cases do we need this, even more it doesn't covered by any test.
Probably, we should modify it or get rid of it at all.

Reply:
Yes, I deleted this piece and. everything continued to work

Reply:
Yes, it should work, however did you understand the reason, why did we need this before, for which cases?

I can't imagine any other examples for now, which will satisfy this condition, instead of `it.responseBody!![0].name`, for which it doesn't work

Reply:
Don't get how it should fix anything?